eng/Subsets.props

```diff
@@ -62,7 +62,8 @@
     <DefaultMonoSubsets Condition="'$(MonoAOTEnableLLVM)' == 'true' and '$(MonoAOTLLVMDir)' == ''">mono.llvm+</DefaultMonoSubsets>
     <DefaultMonoSubsets Condition="'$(TargetOS)' == 'Browser'">$(DefaultMonoSubsets)mono.wasmruntime+</DefaultMonoSubsets>
     <DefaultMonoSubsets Condition="'$(MonoCrossAOTTargetOS)' != ''">$(DefaultMonoSubsets)mono.aotcross+</DefaultMonoSubsets>
-    <DefaultMonoSubsets>$(DefaultMonoSubsets)mono.runtime+mono.corelib+mono.packages+mono.tools+</DefaultMonoSubsets>
+    <DefaultMonoSubsets>$(DefaultMonoSubsets)mono.runtime+mono.corelib+mono.packages+</DefaultMonoSubsets>
+    <DefaultMonoSubsets Condition="'$(PrimaryRuntimeFlavor)' == 'Mono'">$(DefaultMonoSubsets)mono.tools+</DefaultMonoSubsets>
```

Comment:
I think it needs to be `'$(PrimaryRuntimeFlavor)' != 'Mono'`  - if primary runtime flavor is Mono, we do *not* want to build the dotnet-pgo tools.

Reply:
Yes, but only on platforms that also support `CoreCLR`, right?   Or in other words, we want to build dotnet-pgo tool when building with `RuntimeFlavor` equal to `Mono` but `PrimaryRuntimeFlavor` equal to `CoreCLR`.

Since we only get here for `RuntimeFlavor` == `Mono` we only explicitly need to check for the `PrimaryRuntimeFlavor` != `Mono` condition.

Reply:
Ah right, I didn't pay close enough attention 😅 . I agree that it should be changed to the opposite condition.